File: solutions/clojure/resistor-color-duo/1/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def colors-vec
  ["black" "brown" "red" "orange" "yellow" "green" "blue" "violet" "grey" "white"])

(defn resistor-value
  "Returns the resistor value based on the given colors."
  [colors]
  (let [two-bands (take 2 colors)]
    (->>
      two-bands
      (map #(.indexOf colors-vec %))
      clojure.string/join
      Integer/parseInt)))

File: solutions/elixir/resistor-color-duo/2/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """

   @resistor_colors %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    colors
    |> Enum.take(2)
    |> Enum.map(&Map.get(@resistor_colors, &1))
    |> Integer.undigits()
  end
end
